tests/components/blue_current/__init__.py

```diff
@@ -1,42 +1,121 @@
 """Tests for the Blue Current integration."""
 from __future__ import annotations
 
-from unittest.mock import patch
+from asyncio import Event, sleep
+from functools import partial
+from unittest.mock import MagicMock, patch
 
 from bluecurrent_api import Client
 
-from homeassistant.components.blue_current import DOMAIN, Connector
-from homeassistant.config_entries import ConfigEntry
+from homeassistant.components.blue_current import DOMAIN
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.dispatcher import async_dispatcher_send
 
 from tests.common import MockConfigEntry
 
+DEFAULT_CHARGE_POINT = {
+    "evse_id": "101",
+    "model_type": "",
+    "name": "",
+}
+
+
+async def connect(client, token: str):
+    """Set the connected event."""
+    if client.connect_exception:
```

Comment:
I'd just set a different side effect, ie the exception to raise, on the mock connect method instead.